platform/android/src/native_map_view.cpp

```diff
@@ -195,7 +200,15 @@ void NativeMapView::render() {
         mbgl::Log::Info(mbgl::Event::Android, "Not swapping as we are not ready");
     }
 
-    deactivate();
+    if (currentContext != context && currentContext != EGL_NO_CONTEXT) {
+        if (!eglMakeCurrent(currentDisplay, currentDrawSurface, currentReadSurface, currentContext)) {
+            mbgl::Log::Error(mbgl::Event::OpenGL, "eglMakeCurrent() returned error %d",
+                             eglGetError());
+            throw std::runtime_error("eglMakeCurrent() failed");
+        }
+    } else {
+        deactivate();
```

Comment:
@tmpsantos My guess is that  on "Nexus 5 Android 4.4.4", either the View Framework or some other opengl view is changing the opengl state prior to call to NativeMapView::render. Is it possible to do the following experiment in your sandbox with the following two changes?  If the 'screen is white bug'  goes away, its an issue with not handling the opengl state correctly.

1. Instantiate mbgl::Map with GLContextMode::Shared as the fourth argument  [here](https://github.com/mapbox/mapbox-gl-native/blob/master/platform/android/src/native_map_view.cpp#L83). This will ensure that gl state is reset every time by mbgl::Map.
2.  Call glViewPort unconditionally by commenting out if(sizeChanged)  (The current code  will consistently cause crashes/wrong behavior if there are multiple Opengl views in the same application. Its also missing the glError call that MBGL_ERROR_CHECK provides.).